<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <include file="$(find rr_iarrc)/launch/stoplight_watcher.launch"/>
    <include file="$(find rr_iarrc)/launch/race_end_detector.launch"/>

    <include file="$(find rr_iarrc)/launch/urc_lane_follower.launch"/>
    <include file="$(find rr_iarrc)/launch/sign_detector.launch"/>

    <node name="urc_controller" pkg="rr_iarrc" type="urc_controller" output="screen">
        <param name="startSignal" type="string" value="/start_detected"/>
        <param name="resetSignal" type="string" value="/reset_detected"/>
        <param name="turn_speed" type="double" value="0.90"/>
        <param name="left_turn_steering" type="double" value="-0.28"/>
        <param name="right_turn_steering" type="double" value="0.46"/>
        <param name="left_offset" type="double" value="0.0"/>
        <param name="right_offset" type="double" value="0.0"/>
       	<param name="forward_timeout" type="double" value="10.0"/>
    </node>


    <node name="laplacian_line_detector" pkg="rr_iarrc" type="laplacian_line_detector" output="screen">
        <!--Minimum Area to Keep-->
        <param name="perfect_lines_min_cut" type="int" value="50" />
        <!--Strength of Floodfill-->
        <param name="Laplacian_threshold" type="int" value="20" />
        <!--Strength of Adaptive Thresholding-->
        <param name="adaptive_mean_threshold" type="int" value="3" />

        <!--The larger the height the lower it will be on the screen (Image Height = 1280)-->
        <param name="blockSky_height" type="int" value="550" />
        <param name="blockWheels_height" type="int" value="800" />
        <param name="blockBumper_height" type="int" value="650" />

        <param name="subscription_node" type="string" value="/camera_center/image_color_rect"/>
    </node>

    <node pkg="rr_platform" type="image_transform" name="image_transform" output="screen">
        <param name="transform_topics" value="/camera_center/image_color_rect/lines/detection_img"/>
        <param name="px_per_meter" value="50"/>
        <param name="map_dist_max" value="4.0"/>
        <param name="map_dist_min" value="0.0"/>

        <param name="fallback_fov_horizontal" value="1.6955"/>
        <param name="fallback_fov_vertical" value="1.3758"/>
        <param name="fallback_image_width" value="1280"/>
        <param name="fallback_image_height" value="964"/>
    </node>


</launch>
